src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -252,8 +287,93 @@ public ExistingAzureResourceAnnotation(object name, object? resourceGroup = null
         public object? ResourceGroup { get { throw null; } }
     }
 
+    public partial interface IAddRoleAssignmentsContext
+    {
+        AzureResourceInfrastructure Infrastructure { get; }
+
+        global::Azure.Provisioning.BicepValue<System.Guid> PrincipalId { get; }
+
+        global::Azure.Provisioning.BicepValue<string> PrincipalName { get; }
+
+        global::Azure.Provisioning.BicepValue<global::Azure.Provisioning.Authorization.RoleManagementPrincipalType> PrincipalType { get; }
+
+        System.Collections.Generic.IEnumerable<RoleDefinition> Roles { get; }
+    }
+
+    public partial interface IAppIdentityResource
+    {
+        BicepOutputReference ClientId { get; }
+
+        BicepOutputReference Id { get; }
+
+        BicepOutputReference PrincipalId { get; }
+
+        BicepOutputReference PrincipalName { get; }
+    }
+
+    public partial interface IKeyVaultResource : ApplicationModel.IResource, ApplicationModel.IAzureResource
```

Comment:
IAzureKeyVaultResource ? cc @eerhardt

Reply:
It's in the Azure namespace. Does it need to have an `Azure` prefix?

Reply:
No, but the resource is called AzureKeyVaultResource 😄

Reply:
[Rename IKeyVaultResource and IKeyVaultSecretReference (dotnet/aspire#8532)](https://github.com/dotnet/aspire/pull/8532/) renames this